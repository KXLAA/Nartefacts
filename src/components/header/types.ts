export type HeaderProps = {
  primary?: boolean
  secondary?: boolean
}
