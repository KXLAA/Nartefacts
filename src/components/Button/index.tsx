import * as S from './styles'
import { capitalize } from 'utils/utils'

export type ButtonProps = {
  text?: string
  buttonColor?: string
  textColor?: string
}

export const Button: React.FC<ButtonProps> = ({
  buttonColor,
  text,
  textColor,
}) => {
  return (
    <S.Button buttonColor={buttonColor} textColor={textColor} role="button">
      {capitalize(text!) || 'Button'}
    </S.Button>
  )
}
