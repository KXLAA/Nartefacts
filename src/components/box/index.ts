export { Box } from './Box'
export type { BoxProps } from './box.types'
