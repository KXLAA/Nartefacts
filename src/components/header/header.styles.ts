import { styled } from '@/lib/stitches'

export const Text = styled('div', {
  fontSize: 98,
  textDecoration: 'underline',
  fontWeight: 'bold',
})
