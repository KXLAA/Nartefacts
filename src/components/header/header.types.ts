export interface HeaderProps {
  type?: 'primary' | 'secondary' | 'admin'
}
