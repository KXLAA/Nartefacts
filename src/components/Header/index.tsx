import Link from 'next/link'
import * as S from '@/components/Header/styles'

import { Logo } from '@/components/Logo'
import { Button } from '@/components/Button'
import { Spacer } from '@/components/Common/Spacer'

export type HeaderProps = {
  primary?: boolean
  secondary?: boolean
}

export const Header: React.FC<HeaderProps> = ({ primary, secondary }) => {
  return (
    <>
      {primary ? (
        <S.Layout>
          <Logo />

          <S.Nav>
            <Link href="/create" passHref>
              <Button label="create" buttonType="link-primary" width="full" />
            </Link>

            <Link href="/saved" passHref>
              <Button label="saved" buttonType="link-secondary" width="full" />
            </Link>

            <Link href="/info" passHref>
              <Button label="info" buttonType="link-primary" width="full" />
            </Link>
          </S.Nav>
          <Spacer size="lg" />
        </S.Layout>
      ) : null}

      {secondary ? (
        <S.Layout>
          <S.Nav>
            <Link href="/">
              <Logo width="250" />
            </Link>
          </S.Nav>
          <Spacer size="lg" />
        </S.Layout>
      ) : null}
    </>
  )
}
