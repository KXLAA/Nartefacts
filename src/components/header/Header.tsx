import Link from 'next/link'

import { Button } from '@/components/button'
import { Flex } from '@/components/flex'
import { Logo } from '@/components/logo'

import { HeaderProps } from './types'

export const Header = ({ primary, secondary }: HeaderProps) => {
  return (
    <div>
      {primary ? (
        <Flex as="header" direction={'column'} gap={6} justify="between">
          <Logo />
          <Flex as="nav" gap={6}>
            <Link href="/create" passHref>
              <Button variant="dark" label="create" fullWidth />
            </Link>

            <Link href="/saved" passHref>
              <Button label="saved" fullWidth />
            </Link>

            {/* <Link href="/info" passHref>
              <Button label="info" buttonType="link-primary" width="full" />
            </Link> */}
          </Flex>
        </Flex>
      ) : null}

      {secondary ? (
        <Flex as="header">
          <Flex as="nav" justify={'center'}>
            <Link href="/">
              <Logo width="250" />
            </Link>
          </Flex>
        </Flex>
      ) : null}
    </div>
  )
}
