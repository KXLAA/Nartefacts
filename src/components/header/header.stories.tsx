import { Meta, Story } from '@storybook/react'

import { Header } from '@/components/header'
import * as Layout from '@/components/layout'

export default {
  component: Header,
  title: 'Header',
} as Meta

export const Basic: Story = () => (
  <Layout.Main>
    <Header />
  </Layout.Main>
)
