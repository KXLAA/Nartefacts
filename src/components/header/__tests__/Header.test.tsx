import { fireEvent, render } from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'

import { Header } from '@/components/header'
import { createMockRouter } from '@/utils'

describe('<Header />', () => {
  it('should render the header based on props', () => {
    const { getByRole } = render(<Header type="primary" />)
    const header = getByRole('banner')
    expect(header).toBeInTheDocument()
  })

  it('should render <Header /> correctly', () => {
    const { container } = render(<Header type="primary" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the nav', () => {
    const { getByRole } = render(<Header type="primary" />)
    const navigation = getByRole('navigation')
    expect(navigation).toBeInTheDocument()
  })

  it('should render the navigation buttons', () => {
    const { getByText } = render(<Header type="primary" />)
    // const infoLink = getByText(/info/i)
    const savedLink = getByText(/saved/i)
    const createLink = getByText(/create/i)

    // expect(infoLink).toBeInTheDocument()
    expect(savedLink).toBeInTheDocument()
    expect(createLink).toBeInTheDocument()
  })

  it('has navigation buttons that work', () => {
    const router = createMockRouter({
      query: { id: '33' },
      pathname: 'kola',
    })

    const { getByRole } = render(
      <RouterContext.Provider value={router}>
        <Header type="primary" />
      </RouterContext.Provider>,
    )

    const createLink = getByRole('button', { name: /create/i })

    fireEvent.click(createLink)
    expect(router.push).toHaveBeenCalledWith('/create', '/create', {
      locale: undefined,
      scroll: undefined,
      shallow: undefined,
    })
  })
})
