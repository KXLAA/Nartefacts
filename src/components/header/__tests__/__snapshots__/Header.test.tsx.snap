// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render <Header /> correctly 1`] = `
<header
  class="nartefacts-c-PJLV nartefacts-c-fZrNKq nartefacts-c-fZrNKq-iTKOFX-direction-column nartefacts-c-fZrNKq-knmidH-justify-between nartefacts-c-fZrNKq-dCpVsf-gap-6"
>
  <svg
    aria-roledescription="logo"
    class="nartefacts-c-gsmDXe"
    fill="none"
    viewBox="0 0 1206 150"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Logo
    </title>
    <path
      d="M34.0398 37.8H0.439844V146H34.0398V90.2C34.0398 71 46.2398 59.8 65.4398 59.8C84.6398 59.8 89.0399 69.6 89.0399 91.4V146H122.64V85.2C122.64 53 112.44 34 76.2398 34C59.6398 34 45.8398 40.2 34.0398 53V37.8ZM189.854 73.6C189.854 60.8 198.054 55.4 214.454 55.4C231.054 55.4 239.454 59.4 239.454 67.6C239.454 79.6 224.454 78.8 196.654 81C162.254 83.8 145.254 91.6 145.254 114C145.254 135.4 162.854 149.8 193.454 149.8C212.054 149.8 226.454 145.2 238.654 134.2L240.254 146H277.254V143.6C273.254 140.8 272.254 136 272.254 128.6V74.8C272.254 40.6 245.654 34 216.454 34C176.054 34 155.054 48 155.054 73.6H189.854ZM239.454 93.8V97.6C239.454 114.8 224.254 127 201.254 127C190.054 127 183.054 121.4 183.054 113.2C183.054 102.8 193.054 101.4 212.054 99.4C223.254 98.2 232.454 97.6 239.454 93.8ZM333.454 37.8H299.854V146H333.454V97.6C333.454 75.8 345.454 62.8 365.054 62.8C368.854 62.8 373.854 63 375.654 63.2V34.8C373.054 34.4 368.654 34 366.254 34C354.054 34 343.454 40 333.454 52V37.8ZM399.86 8.99999V37.8H384.06V63.6H399.86V123.6C399.86 143 411.26 149.8 435.86 149.8H458.66V124H444.86C433.66 124 433.46 121.6 433.46 109V63.6H458.66V37.8H433.46V8.99999H399.86ZM598.03 110.8H561.03C557.03 120.8 547.43 126.6 534.63 126.6C518.63 126.6 507.23 116.4 507.23 100.6H599.83V96C599.83 56.2 576.43 34 534.63 34C494.03 34 469.43 56.4 469.43 91.8C469.43 127.4 494.43 149.8 534.63 149.8C566.63 149.8 589.03 135.4 598.03 110.8ZM508.23 78.2C509.43 64.8 519.83 56.4 535.43 56.4C551.03 56.4 561.43 64.8 562.63 78.2H508.23ZM687.957 0.799998H665.157C640.557 0.799998 629.157 7.4 629.157 26.8V37.8H613.357V63.6H629.157V146H662.757V63.6H687.957V37.8H662.757V35.2C662.757 28.2 665.357 26.6 673.957 26.6H687.957V0.799998ZM745.127 73.6C745.127 60.8 753.327 55.4 769.727 55.4C786.327 55.4 794.727 59.4 794.727 67.6C794.727 79.6 779.727 78.8 751.927 81C717.527 83.8 700.527 91.6 700.527 114C700.527 135.4 718.127 149.8 748.727 149.8C767.327 149.8 781.727 145.2 793.927 134.2L795.527 146H832.527V143.6C828.527 140.8 827.527 136 827.527 128.6V74.8C827.527 40.6 800.927 34 771.727 34C731.327 34 710.327 48 710.327 73.6H745.127ZM794.727 93.8V97.6C794.727 114.8 779.527 127 756.527 127C745.327 127 738.327 121.4 738.327 113.2C738.327 102.8 748.327 101.4 767.327 99.4C778.527 98.2 787.727 97.6 794.727 93.8ZM979.927 79.4C976.327 50.4 954.527 34 917.327 34C874.327 34 848.727 56.4 848.727 91.8C848.727 127 875.127 149.8 914.927 149.8C953.127 149.8 978.927 130 981.127 102H943.127C942.127 115 930.727 124 915.327 124C897.127 124 885.127 111.4 885.127 92C885.127 72.4 897.327 59.8 915.327 59.8C931.327 59.8 941.127 66.8 942.927 79.4H979.927ZM1009.82 8.99999V37.8H994.021V63.6H1009.82V123.6C1009.82 143 1021.22 149.8 1045.82 149.8H1068.62V124H1054.82C1043.62 124 1043.42 121.6 1043.42 109V63.6H1068.62V37.8H1043.42V8.99999H1009.82ZM1202.19 70C1202.19 47.6 1181.39 34 1144.19 34C1105.59 34 1085.39 47.2 1085.39 70.6C1085.39 88.8 1098.99 101.8 1121.59 103.8L1151.99 106.4C1164.79 107.4 1170.79 108.6 1170.79 115.4C1170.79 121.2 1162.39 126 1147.59 126C1127.99 126 1118.19 121.8 1117.19 110.4H1081.79C1083.39 136.4 1104.39 149.8 1144.79 149.8C1184.99 149.8 1205.59 135.8 1205.59 110.2C1205.59 91.4 1192.79 79.8 1169.39 78L1145.79 76.2C1128.79 74.8 1120.59 72.2 1120.59 65.8C1120.59 60 1128.19 56 1142.39 56C1158.39 56 1167.19 60.4 1168.79 70H1202.19Z"
      fill="#5A5A5A"
    />
  </svg>
  <nav
    class="nartefacts-c-PJLV nartefacts-c-fZrNKq nartefacts-c-fZrNKq-ejCoEP-direction-row nartefacts-c-fZrNKq-kdofoX-gap-2 nartefacts-c-fZrNKq-kgtBHb-gap-6"
  >
    <button
      class="nartefacts-c-hjTOsc nartefacts-c-hjTOsc-bgGQWB-size-sm nartefacts-c-hjTOsc-eVtbBr-variant-dark nartefacts-c-hjTOsc-deuXAd-fullWidth-true nartefacts-c-hjTOsc-iciwRX-size-lg nartefacts-c-hjTOsc-iPJLV-css"
      href="/create"
    >
      create
    </button>
    <button
      class="nartefacts-c-hjTOsc nartefacts-c-hjTOsc-bgGQWB-size-sm nartefacts-c-hjTOsc-eDnYwL-variant-light nartefacts-c-hjTOsc-deuXAd-fullWidth-true nartefacts-c-hjTOsc-iciwRX-size-lg nartefacts-c-hjTOsc-iPJLV-css"
      href="/saved"
    >
      saved
    </button>
  </nav>
</header>
`;
